### PR TITLE
Fix model sets example

### PR DESCRIPTION
**Pull request by nden**
### Description

This pull request supersedes #9283 and includes a comprehensive example of fitting a model set to data.
I can't force push to the original PR but I kept the commits and fixed the failures.

@karllark Please review.

### CODE REVIEW

**Comment by pllim** [comment]
As discussed offline, I will step in to do official approval when I get an approval from @karllark .

**Comment by karllark** [comment]
Looks good to me and ready to merge.

**Comment by pllim** [comment]
Wait, wait... I might have seen a formatting typo... let me double check at the rendered doc real quick... ✋

File: docs/modeling/example-fitting-model-sets.rst [hunk 1]
```diff
@@ -1,37 +1,173 @@
...
+    >>> image.shape
+    (10, 3, 4)
+
+Create the models and the fitter. We need N=width*height instances of the same linear,
+parametric model (model sets currently only work with linear models and fitters)::
+
+    >>> N = width * height
+    >>> line = models.Polynomial1D(degree=1, n_models=N)
+    >>> fit = fitting.LinearLSQFitter()
+    >>> print("We created %d models" % len(line))
```
Comment (pllim) [inline docs/modeling/example-fitting-model-sets.rst]: Nitpick: f-string
```suggestion
    >>> print(f"We created {len(line)} models")
```

File: docs/modeling/example-fitting-model-sets.rst [hunk 2]
```diff
@@ -1,37 +1,173 @@
...
+    >>> y = pixels.T
+    >>> print("x axis is one dimensional: ",t.shape)
+    x axis is one dimensional:  (10,)
+    >>> print("y axis is two dimensional, N by len(x): ", y.shape)
+    y axis is two dimensional, N by len(x):  (12, 10)
+
+Fit the model. It fits the N models simultaneously::
+
+    >>> new_model = fit(line, x=t, y=y)
+    >>> print("We fit %d models" % len(new_model))
```
Comment (pllim) [inline docs/modeling/example-fitting-model-sets.rst]:
```suggestion
    >>> print(f"We fit {len(new_model)} models")
```

File: docs/modeling/example-fitting-model-sets.rst [hunk 3]
```diff
@@ -1,37 +1,173 @@
...
+        -4.7258541419613165 10.315295021908833
+           4.95631951675311 10.911167956770575
+
+    >>> print("The new_model has a param_sets attribute with shape: ",new_model.param_sets.shape)
+    The new_model has a param_sets attribute with shape:  (2, 12)
+
+    >>> print("And values that are the best-fit parameters for each pixel: ")
+    And values that are the best-fit parameters for each pixel:
+
+    >>> print(new_model.param_sets) # doctest: +FLOAT_CMP
```
Comment (pllim) [inline docs/modeling/example-fitting-model-sets.rst]: Makes more sense to combine these two print functions, narrative-wise:
```suggestion
    >>> print(f"And values that are the best-fit parameters for each pixel:\n{new_model.param_sets}") # doctest: +FLOAT_CMP
```

File: docs/modeling/example-fitting-model-sets.rst [hunk 4]
```diff
@@ -1,37 +1,173 @@
...
+        -5.8828309622531565  7.016044775363114
+        -11.676538736037775  8.072519832452022
+          -6.17932185981594  9.103924115403503
+        -4.7258541419613165 10.315295021908833
+           4.95631951675311 10.911167956770575
+
+    >>> print("The new_model has a param_sets attribute with shape: ",new_model.param_sets.shape)
+    The new_model has a param_sets attribute with shape:  (2, 12)
+
+    >>> print("And values that are the best-fit parameters for each pixel: ")
```
Comment (pllim) [inline docs/modeling/example-fitting-model-sets.rst]:
```suggestion
```

File: docs/modeling/example-fitting-model-sets.rst [hunk 5]
```diff
@@ -1,37 +1,173 @@
...
+    And values that are the best-fit parameters for each pixel:
```
Comment (pllim) [inline docs/modeling/example-fitting-model-sets.rst]:
```suggestion
```

File: docs/modeling/example-fitting-model-sets.rst [hunk 6]
```diff
@@ -1,37 +1,173 @@
...
+
```
Comment (pllim) [inline docs/modeling/example-fitting-model-sets.rst]:
```suggestion
```

File: docs/modeling/example-fitting-model-sets.rst [hunk 7]
```diff
@@ -1,37 +1,173 @@
...
+        6.78222337   3.6282205   -5.88283096 -11.67653874  -6.17932186
+       -4.72585414   4.95631952]
+     [  0.           1.04639983   1.98187335   3.04927988   4.01381043
+        4.75591271   5.84139795   7.01604478   8.07251983   9.10392412
+       10.31529502  10.91116796]]
+
+Plot the fit along a couple of pixels:
+
+    >>> def plotramp(t, image, best_fit, row, col):
+    ...     plt.plot(t, image[:, row, col], '.', label='data pixel %d,%d' % (row, col))
```
Comment (pllim) [inline docs/modeling/example-fitting-model-sets.rst]:
```suggestion
    ...     plt.plot(t, image[:, row, col], '.', label=f'data pixel {row},{col}')
```

File: docs/modeling/example-fitting-model-sets.rst [hunk 8]
```diff
@@ -1,37 +1,173 @@
...
+    ...     plt.plot(t, best_fit[:, row, col], '-', label='fit to pixel %d,%d' % (row, col))
```
Comment (pllim) [inline docs/modeling/example-fitting-model-sets.rst]:
```suggestion
    ...     plt.plot(t, best_fit[:, row, col], '-', label=f'fit to pixel {row},{col}')
```

File: docs/modeling/example-fitting-model-sets.rst [hunk 9]
```diff
@@ -1,37 +1,173 @@
...
+    # Fit the model. It does the looping over the N models implicitly
+    new_model = fit(line, x=t, y=y)
+
+    # Fill an array with values computed from the best fit and reshape it to match the original
+    best_fit = new_model(t, model_set_axis=False).T.reshape((depth, height, width))
+
+
+    # Plot the fit along a couple of pixels
+    def plotramp(t, image, best_fit, row, col):
+        plt.plot(t, image[:, row, col], '.', label='data pixel %d,%d' % (row, col))
```
Comment (pllim) [inline docs/modeling/example-fitting-model-sets.rst]:
```suggestion
        plt.plot(t, image[:, row, col], '.', label=f'data pixel {row},{col}')
```

File: docs/modeling/example-fitting-model-sets.rst [hunk 10]
```diff
@@ -1,37 +1,173 @@
...
+        plt.plot(t, best_fit[:, row, col], '-', label='fit to pixel %d,%d' % (row, col))
```
Comment (pllim) [inline docs/modeling/example-fitting-model-sets.rst]:
```suggestion
        plt.plot(t, best_fit[:, row, col], '-', label=f'fit to pixel {row},{col}')
```

File: docs/modeling/example-fitting-model-sets.rst [hunk 11]
```diff
@@ -1,37 +1,173 @@
...
+           4.95631951675311 10.911167956770575
+
+    >>> print("The new_model has a param_sets attribute with shape: ",new_model.param_sets.shape)
+    The new_model has a param_sets attribute with shape:  (2, 12)
+
+    >>> print("And values that are the best-fit parameters for each pixel: ")
+    And values that are the best-fit parameters for each pixel:
+
+    >>> print(new_model.param_sets) # doctest: +FLOAT_CMP
+    [[  0.          -0.52066063   0.64019304   0.1134713   -3.35564204
```
Comment (pllim) [inline docs/modeling/example-fitting-model-sets.rst]: Ops, forgot to move the output...
```suggestion
    And values that are the best-fit parameters for each pixel:
    [[  0.          -0.52066063   0.64019304   0.1134713   -3.35564204
```

File: docs/modeling/example-fitting-model-sets.rst (hunk 4 above)
Comment (pllim) [reply docs/modeling/example-fitting-model-sets.rst]: I think you missed this one... `doctest` will probably fail. 🤷‍♀

**Pull request merged**